### PR TITLE
CORE-18169: Change User password - db worker changes

### PR DESCRIPTION
**Pull request by Tom-Fitzpatrick**
This PR implements the db-worker side of the User password change feature in `UserWriterImpl.kt`

Because we can get the username from the request header, this also changes the endpoints: 
- `{loginName}/otheruserpassword` to `/otheruserpassword`
- `{loginName}/selfpassword` to `/selfpassword`

This adds change password methods to ClusterBuilder to support future e2e tests.

Corda-api changes to support this: https://github.com/corda/corda-api/pull/1418

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -44,7 +44,7 @@ commonsLangVersion = 3.12.0
 commonsTextVersion = 1.10.0
 # Corda API libs revision (change in 4th digit indicates a breaking change)
 # Change to 5.2.0.xx-SNAPSHOT to pick up maven local published copy
-cordaApiVersion=5.2.0.20-beta+
+cordaApiVersion=5.2.0.21-alpha-1703075132231
```
Comment (Tom-Fitzpatrick) [inline gradle.properties]: I will update this to point to the beta, once the corresponding corda-api PR is merged
https://github.com/corda/corda-api/pull/1418

File: libs/permissions/permission-storage-writer-impl/src/main/kotlin/net/corda/libs/permissions/storage/writer/impl/user/impl/UserWriterImpl.kt [hunk 2]
```diff
@@ -43,6 +44,39 @@ class UserWriterImpl(
...
+        request: ChangeUserPasswordRequest,
+        requestUserId: String
+    ): net.corda.data.permissions.User {
+        log.debug { "Received request to change password for user: ${request.requestedBy}" }
+        return entityManagerFactory.transaction { entityManager ->
+
+            val validator = EntityValidationUtil(entityManager)
+            val user = validator.validateAndGetUniqueUser(request.requestedBy)
+
+            println("old password hash: ${user.hashedPassword}")
```
Comment (vkolomeyko) [inline libs/permissions/permission-storage-writer-impl/src/main/kotlin/net/corda/libs/permissions/storage/writer/impl/user/impl/UserWriterImpl.kt]:
```suggestion
            log.debug { "old password hash: ${user.hashedPassword}" }
```

File: gradle.properties (hunk 1 above)
Comment (vkolomeyko) [inline gradle.properties]: - [x] Will need to revert this change once `corda-api` PR is merged.

**Review by vkolomeyko — COMMENTED** [review]
Looks fine to me, minor comment posted.

File: libs/permissions/permission-storage-writer-impl/src/main/kotlin/net/corda/libs/permissions/storage/writer/impl/user/impl/UserWriterImpl.kt (hunk 2 above)
Comment (Tom-Fitzpatrick) [reply libs/permissions/permission-storage-writer-impl/src/main/kotlin/net/corda/libs/permissions/storage/writer/impl/user/impl/UserWriterImpl.kt]: Ah thanks, I'll just remove this actually as it was left over from when I was debugging it

File: gradle.properties (hunk 1 above)
Comment (Tom-Fitzpatrick) [reply gradle.properties]: Merged the corda-api PR and updated this config

**Review by vkolomeyko — APPROVED** [review]
LGTM
